### PR TITLE
fix: regression when calculating params and pattern

### PR DESCRIPTION
**Pull request by ematipico**
## Changes

Closes https://github.com/withastro/astro/issues/9041
Closes https://github.com/withastro/astro/issues/9040

The issue comes from a naive refactor of mine when coding i18n stuff.

I reverted the code.

## Testing

I added a new test that failed at the beginning, and now it passes

## Docs

N/A

### CODE REVIEW

File: packages/astro/src/core/routing/manifest/create.ts
Comment (ematipico) [inline packages/astro/src/core/routing/manifest/create.ts:1]: This is the file that contains the issue, I reverted the signature of `getPattern`

**Comment by Princesseuh** [comment]
I assume this also fixes https://github.com/withastro/astro/issues/9040, right?

**Comment by ematipico** [comment]
> I assume this also fixes #9040, right?

Yes :) updated the PR description

**Review by Princesseuh — APPROVED** [review]

File: packages/integrations/sitemap/test/staticPaths.test.js [hunk 1]
```diff
@@ -30,7 +31,8 @@ describe('getStaticPaths support', () => {
 		expect(urls).to.not.include('http://example.com/de/404/');
 	});
 
-	it('includes numerical pages', () => {
-		expect(urls).to.include('http://example.com/123/');
+	it('includes numerical pages', async () => {
+		const page = await fixture.readFile('./it/manifest');
+		expect(page).to.contain('I\'m a route in the "it" language.');
 	});
```
Comment (bluwy) [inline packages/integrations/sitemap/test/staticPaths.test.js:37]: Should the old test be kept? The new changes replaces it.

Comment (ematipico) [reply packages/integrations/sitemap/test/staticPaths.test.js:37]: Good Catch, I'll restore the previous test

**Review by bluwy — APPROVED** [review]

**Comment by itsmatteomanf** [comment]
@ematipico note that this broke the root to default locale redirect, it doesn't respect the setting and does redirect to no trailing slash even with `trailingSlash: 'always'`.

**Pull request merged**